src/lib.rs

```diff
@@ -115,9 +117,9 @@ impl EncodingDetector {
     //    unsafe { ffi::uchardet_reset(self.ptr); }
     //}
 
-    /// Get the decoder's current best guess as to the encoding.  Returns
+    /// Get the decoder's current best guess as to the encoding. Returns
     /// `None` on error, or if the data appears to be ASCII.
-    fn charset(&self) -> Option<String> {
+    fn charset(&self) -> Result<String, EncodingDetectorError> {
```

Comment:
This type is written as `EncodingDetectorResult<String>` above, but `Result<String, EncodingDetectorError>` here. At the least we should be consistent—but I'm halfway tempted to convert this library to use `error-chain` since we're making breaking API changes, and since our error type is relatively heavyweight already thanks to the `message: String` field. What do you think?

I'm totally happy to handle the `error-chain` conversion myself; I've done several of them lately.